sklearn/neighbors/graph.py

```diff
@@ -77,7 +79,13 @@ def kneighbors_graph(X, n_neighbors, mode='connectivity', metric='minkowski',
             ).fit(X)
     else:
         _check_params(X, metric, p, metric_params)
-    return X.kneighbors_graph(X._fit_X, n_neighbors, mode=mode)
+    if mode == "connectivity":
+        warnings.warn("The behavior of marking each sample itself as the "
```

Comment:
Ok, let's not use the word deprecated. But I am still not sure on what a good plan of action is. We want to keep this method and have it be useful in the future, right? And we don't want to change current behavior. But it should also be useful now. If it is not useful now we can basically deprecate the whole function and remove it in the future, because everybody needs to switch away from it because it is broken.
So what I would like to have is a future-proof way for people to use the fixed version.